packages/react-query/src/useQueries.ts

```diff
@@ -188,5 +190,35 @@ export function useQueries<T extends any[]>({
     observer.setQueries(defaultedQueries, { listeners: false })
   }, [defaultedQueries, observer])
 
+  const errorResetBoundary = useQueryErrorResetBoundary()
+
+  defaultedQueries.forEach((query) => {
+    if (query.suspense || query.useErrorBoundary) {
+      // Prevent retrying failed query if the error boundary has not been reset yet
+      if (!errorResetBoundary.isReset()) {
+        query.retryOnMount = false
+      }
+    }
+  })
+
+  React.useEffect(() => {
+    errorResetBoundary.clearReset()
+  }, [errorResetBoundary])
+
+  const firstSingleResultWhichShouldThrow = result.find(
+    (singleResult, index) =>
+      singleResult.isError &&
+      !errorResetBoundary.isReset() &&
+      !singleResult.isFetching &&
+      shouldThrowError(defaultedQueries[index]?.useErrorBoundary ?? false, [
+        singleResult.error,
+        observer.getQueries()[index]!,
+      ]),
```

Comment:
I also think that just this part could be a single extracted util function

Reply:
[7f81e88](https://github.com/TanStack/query/pull/4177/commits/7f81e8893bd674f103d6b72a93b77151abea86a3)